tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -243,16 +244,44 @@ struct UnsortedSegmentSumFunctor<CPUDevice, T, Index> {
     }
   }
 };
-
+// UnsortedSegmentMaxFunctor implementation for CPUDevice.
+template <typename T, typename Index>
+struct UnsortedSegmentMaxFunctor<CPUDevice, T, Index>
+    : UnsortedSegmentBaseFunctor<CPUDevice, T, Index> {
+  void operator()(OpKernelContext* ctx, const CPUDevice& d,
+                  const Index output_rows, const TensorShape& segment_ids_shape,
+                  typename TTypes<Index>::ConstFlat segment_ids,
+                  const Index data_size, const T* data,
+                  typename TTypes<T, 2>::Tensor output) override {
+    output.setConstant(std::numeric_limits<T>::min());
```

Comment:
So I've been looking at this problem for a while now and I don't really see any straight forward solution to this.
One thing that makes it tricky is the GPU implementation of UnsortedSegmentSum, which does not share the code of the CPU implementation.
So that would be different from baseFunctor, but would need the same signature, since its used as well in BaseOp.
Thus would contain the init functors and reduction functors but would not use them.

Second I can't seem to initialize a functor as a member in e.g. UnsortedSegmentSumOp with member init functors and reduction functors as arguments.
If you see another way I'd be happy if you let me know.